### PR TITLE
HDDS-4201. Improve the performance of OmKeyLocationInfoGroup

### PR DESCRIPTION
**Pull request by runzhiwang**
## What changes were proposed in this pull request?

   ` locations.stream().collect(Collectors.groupingBy(OmKeyLocationInfo::getCreateVersion))` cost 2-3ms even though the locations size is zero.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4201

## How was this patch tested?

no need to add test.

### CODE REVIEW

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyLocationInfoGroup.java [hunk 1]
```diff
@@ -36,8 +36,13 @@
...
                                 List<OmKeyLocationInfo> locations) {
     this.version = version;
-    this.locationVersionMap = locations.stream()
-        .collect(Collectors.groupingBy(OmKeyLocationInfo::getCreateVersion));
+    locationVersionMap = new HashMap<>();
+    for (OmKeyLocationInfo info : locations) {
+      if (!locationVersionMap.containsKey(info.getCreateVersion())) {
+        locationVersionMap.put(info.getCreateVersion(), new ArrayList<>());
+      }
+      locationVersionMap.get(info.getCreateVersion()).add(info);
```
Comment (adoroszlai) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyLocationInfoGroup.java:44]: I think we can avoid two map lookups (and long autoboxing) by using `computeIfAbsent`:

```suggestion
      locationVersionMap
          .computeIfAbsent(info.getCreateVersion(), v -> new ArrayList<>())
          .add(info);
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @runzhiwang for working on this.

**Comment by runzhiwang** [comment]
@adoroszlai Thanks for review. I have updated the patch.

**Review by adoroszlai — APPROVED** [review]
Thanks @runzhiwang for updating the patch.

**Pull request merged**